# What is google colab ?
# Triger google colab app on your google drive
# Mounting your google drive
# Finding the path to a given directory in your drive
# Install yahoo finance library
# import yfinance and use it to get historical priice data
# Working with downloaded data

# Go to your BB and clik on

How to Use Google Colab




## To activate google colab,
### (i) log on to my.fordham.edu
### (ii) click on 'G' and hol and then click on the triangle sign next to the email icon
### Your Google drive can act like a computer. You are probable familer with (i) Google Doc; (ii) Google Sheet ; and (iii) Google slides

### Click on the plus sign and you will see the abve apps. If you click more you find Google Colaboratory. Clickin on activates this app when you use it for the first time


##  Mounting your drive

## On your computer you can navigate your files depending on the operating system on your computer: Windows or Mac

## There is an operating system library we can use to list files in your working directory.

### As we import this library and use it you will learn about its capabilities.

In [1]:
import os
os.getcwd() # this gets the working directory

'/content'

In [2]:
os.listdir()

['.config', 'sample_data']

## The above is a folder on your Google drive. It gets created once you activate google colab

## Mounting the drive into python allows us to navigate files and specify your working directory and specify file paths

## The following mounts your google drive to be access by google colab python

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## This will get you three security steps by wich you allow python to access your google drive

In [4]:
os.getcwd()

'/content'

## So far the working directory is still the folder contents

In [5]:
wd = '/content/drive/MyDrive/Computational finance/Fall-2025/lecture-20250925'
os.chdir(wd)

In [6]:
os.getcwd()

'/content/drive/MyDrive/Computational finance/Fall-2025/lecture-20250925'

## we can list all files in this folders

In [7]:
os.listdir()

['gcolab0_YS.ipynb']

In [9]:
!ls

gcolab0_YS.ipynb


# Install yfinance

In [10]:
!pip install yfinance

In [13]:
import pandas as pd
import yfinance as yf

In [14]:
yf.download('NVDA', period="max")

/tmp/ipython-input-2707956864.py:1: FutureWarning: YF.download() has changed argument auto_adjust default to True
  yf.download('NVDA', period="max")
[*********************100%***********************]  1 of 1 completed


Price            Close        High         Low        Open      Volume
Ticker            NVDA        NVDA        NVDA        NVDA        NVDA
Date                                                                  
1999-01-22    0.037607    0.044770    0.035577    0.040114  2714688000
1999-01-25    0.041547    0.042024    0.037607    0.040591   510480000
1999-01-26    0.038323    0.042860    0.037726    0.042024   343200000
1999-01-27    0.038204    0.039398    0.036293    0.038442   244368000
1999-01-28    0.038084    0.038442    0.037845    0.038204   227520000
...                ...         ...         ...         ...         ...
2025-09-18  176.240005  177.100006  172.960007  173.979996   191763300
2025-09-19  176.669998  178.080002  175.179993  175.770004   237182100
2025-09-22  183.610001  184.550003  174.710007  175.300003   269637000
2025-09-23  178.429993  182.419998  176.210007  181.970001   192559600
2025-09-24  176.970001  179.779999  175.399994  179.720001   142821978

[6710 rows x 5 columns]

In [15]:
yf.download('nvda', period="max")

/tmp/ipython-input-2770226274.py:1: FutureWarning: YF.download() has changed argument auto_adjust default to True
  yf.download('nvda', period="max")
[*********************100%***********************]  1 of 1 completed


Price            Close        High         Low        Open      Volume
Ticker            NVDA        NVDA        NVDA        NVDA        NVDA
Date                                                                  
1999-01-22    0.037607    0.044770    0.035577    0.040114  2714688000
1999-01-25    0.041547    0.042024    0.037607    0.040591   510480000
1999-01-26    0.038323    0.042860    0.037726    0.042024   343200000
1999-01-27    0.038204    0.039398    0.036293    0.038442   244368000
1999-01-28    0.038084    0.038442    0.037845    0.038204   227520000
...                ...         ...         ...         ...         ...
2025-09-18  176.240005  177.100006  172.960007  173.979996   191763300
2025-09-19  176.669998  178.080002  175.179993  175.770004   237182100
2025-09-22  183.610001  184.550003  174.710007  175.300003   269637000
2025-09-23  178.429993  182.419998  176.210007  181.970001   192559600
2025-09-24  176.970001  179.779999  175.399994  179.720001   142821978

[6710 rows x 5 columns]

In [ ]:
symbol = 'nvda'

In [16]:
nvida_daily = yf.download('nvda', period="max")
type(nvida_daily)

/tmp/ipython-input-1115863796.py:1: FutureWarning: YF.download() has changed argument auto_adjust default to True
  nvida_daily = yf.download('nvda', period="max")
[*********************100%***********************]  1 of 1 completed


pandas.core.frame.DataFrame

In [17]:
nvida_daily.head()

Price          Close      High       Low      Open      Volume
Ticker          NVDA      NVDA      NVDA      NVDA        NVDA
Date                                                          
1999-01-22  0.037607  0.044770  0.035577  0.040114  2714688000
1999-01-25  0.041547  0.042024  0.037607  0.040591   510480000
1999-01-26  0.038323  0.042860  0.037726  0.042024   343200000
1999-01-27  0.038204  0.039398  0.036293  0.038442   244368000
1999-01-28  0.038084  0.038442  0.037845  0.038204   227520000

In [18]:
nvida_daily.columns

MultiIndex([( 'Close', 'NVDA'),
            (  'High', 'NVDA'),
            (   'Low', 'NVDA'),
            (  'Open', 'NVDA'),
            ('Volume', 'NVDA')],
           names=['Price', 'Ticker'])

In [ ]:
## Note that column names ara a multiIndex

In [19]:
type( nvida_daily.columns )

pandas.core.indexes.multi.MultiIndex

## I can access the first livel index

In [21]:
nvida_daily.columns[0]

('Close', 'NVDA')

##  Is this a list?


In [ ]:
## It is actually a tupple

In [23]:
nvida_daily.columns[0][0]

'Close'

## For convenience I can suppres the ticker in the column names  

In [24]:
nvida_daily.columns.get_level_values(0)

Index(['Close', 'High', 'Low', 'Open', 'Volume'], dtype='object', name='Price')

In [26]:
nvida_daily.columns = nvida_daily.columns.get_level_values(0)
nvida_daily.head()

Price          Close      High       Low      Open      Volume
Date                                                          
1999-01-22  0.037607  0.044770  0.035577  0.040114  2714688000
1999-01-25  0.041547  0.042024  0.037607  0.040591   510480000
1999-01-26  0.038323  0.042860  0.037726  0.042024   343200000
1999-01-27  0.038204  0.039398  0.036293  0.038442   244368000
1999-01-28  0.038084  0.038442  0.037845  0.038204   227520000

## Discuss adjusted-close and adjusted prices

In [28]:
nvida_daily.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 6710 entries, 1999-01-22 to 2025-09-24
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Close   6710 non-null   float64
 1   High    6710 non-null   float64
 2   Low     6710 non-null   float64
 3   Open    6710 non-null   float64
 4   Volume  6710 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 314.5 KB


## What is the daily growth in NVDA stock since inception

In [31]:
nvida_daily.shape

(6710, 5)

In [32]:
nvida_daily.shape[0]

6710

In [ ]:
## yahoo finance provide data for business days only.
## How many yeas we have in 6710 days?

In [33]:
nvida_daily.shape[0]/252

26.626984126984127

In [ ]:
## If you invested a $1 in Nvidia 1999-01-22 and held it tel 2025-09-24 What is you accumulated investment?

In [37]:
TV = nvida_daily['Close'][-1]/nvida_daily['Close'][0]
round(TV,6)


/tmp/ipython-input-2544929998.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  TV = nvida_daily['Close'][-1]/nvida_daily['Close'][0]


np.float64(4705.776247)

In [ ]:
## Annual-growth is

In [45]:
ndays = nvida_daily.shape[0]
print ( 'ndays= ' , round( ndays/252,2) )
print ( 'daily growth' , round  (TV**(1/ndays),6 ) -1)

ndays=  26.63
daily growth 0.0012609999999999566


In [48]:
print ( 'daily growth' , round  (TV**(252/ndays),6 )-1)

daily growth 0.37381700000000007


## Annual return of 37.38%

# This is very large. Very few stocks provided such return

In [ ]:
## create a column tha calculates the return on the IPO date

In [ ]:
## create a column tha calculates the return on the IPO week of the ipo

In [51]:
from datetime import datetime
nvida_daily.index[0]

Timestamp('1999-01-22 00:00:00')

In [52]:
nvida_daily.index[0].weekday()

4

## This is the 4th business day of the week. It is fRiday

In [53]:
nvida_daily.index[0].day_name()

'Friday'

In [54]:
nvida_daily.index[4].day_name()

'Thursday'

## Calculate the return of the first week from from IPO open to next Thursday close

In [57]:
round ( nvida_daily['Close'][4] / nvida_daily['Open'][0] -1 , 6)

/tmp/ipython-input-4196291624.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  round ( nvida_daily['Close'][4] / nvida_daily['Open'][0] -1 , 6)


np.float64(-0.050606)